### PR TITLE
Feat: Next Server and Client

### PR DESCRIPTION
**Pull request by Emmanuel-Develops**

### CODE REVIEW

File: src/hooks/useGlobalHotkey.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { useHotkeys } from "react-hotkeys-hook";
 import { isMac } from "../utils/userOS";
+// import useIsMac from "@/hooks/useIsMac";
```
Comment (Extheoisah) [inline src/hooks/useGlobalHotkey.ts:3]: remove

File: src/hooks/useIsInitialStateWithoutFilter.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  if (
+    resultLength && (searchQuery || hasFilters)
+  ) {
+    hiddenBody = false;
+  } else {
+    hiddenHomeFacet = false
+  }
+
+  
+  // if (state) {
```
Comment (Extheoisah) [inline src/hooks/useIsInitialStateWithoutFilter.ts:26]: remove if not needed

File: src/hooks/useIsMac.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { useEffect, useRef } from 'react'
+
+const useIsMac = () => {
```
Comment (Extheoisah) [inline src/hooks/useIsMac.ts:3]: can't find where we're using this hook.

File: src/pages/api/elasticSearchProxy/autocomplete.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (Extheoisah) [inline src/pages/api/elasticSearchProxy/autocomplete.ts:1]: we can delete this file if its not in use

File: src/pages/api/elasticSearchProxy/connector.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import { QueryConfig, RequestState } from "@elastic/search-ui";
+import ElasticsearchAPIConnector from "@elastic/search-ui-elasticsearch-connector";
+
+// const connector = new ElasticsearchAPIConnector({
```
Comment (Extheoisah) [inline src/pages/api/elasticSearchProxy/connector.ts:4]: remove if not needed

File: src/pages/api/elasticSearchProxy/search.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  let searchQuery = buildQuery({queryString, filterFields, sortFields, from, size });
+
+  try {
+    // Call the search method
+    const result = await client.search({
+      index: process.env.INDEX,
+      ...searchQuery,
+    });
+    
+    // console.log("🚀 ~ file: search.ts:37 ~ handler ~ result:", result.hits.total)
```
Comment (Extheoisah) [inline src/pages/api/elasticSearchProxy/search.ts:31]: remove console logs

**Review by Extheoisah — CHANGES_REQUESTED** [review]
Quite a big pr. A lot of commented and unused code bit looking good otherwise.

File: src/components/customResults/CustomResults.jsx [hunk 7]
```diff
@@ -1,57 +1,76 @@
...
-        formattedResults.push([{ ...result }]);
-        similarity[locatorId] = idx;
-      }
-    } else {
-      formattedResults.push([{ ...result }]);
-    }
-  });
-
-  sortGroupedResults(groupedIndices, formattedResults);
+  // results.forEach((result) => {
```
[the remark this answers is not in the document]
Comment (Emmanuel-Develops) [reply src/components/customResults/CustomResults.jsx:38]: there's a grouping i did previously, still debating if that's necessary

**Comment by Emmanuel-Develops** [comment]
@Extheoisah, yeah there is a lot of cleaning up to do as I mentioned earlier.
Wanted to get functionality out of the way first

File: src/components/customResults/Thumbnail.jsx [hunk 8]
```diff
@@ -11,7 +13,7 @@ export const Thumbnail = ({ url }) => {
   return (
     <a href={url} target="_blank" rel="noreferrer">
       <div className="search-result-media">
-        <img src={thumbnailURL} />
+        <img src={thumbnailURL} alt="thumbnail" />
```
[the remark this answers is not in the document]
Comment (Emmanuel-Develops) [reply src/components/customResults/Thumbnail.jsx:16]: nextImage would want us to specify the domains in the `next.config.js`. It allows a maximum of 50 domains while we currently have 59.
I could use a wildcard to catch all domains but that defeats the purpose and it is not recommended

File: src/components/customResults/Result.jsx [hunk 9]
```diff
@@ -30,6 +33,29 @@ const Result = ({
...
+            .map((i) => i.text)
+            .join(" ")
+        } catch {
+          return body || result.body_formatted
+        }
+      }
+    }
+  }
+
+  // removed onClickLink
```
[the remark this answers is not in the document]
Comment (Extheoisah) [reply src/components/customResults/Result.jsx:58]: Just this remaining and we can merge

**Pull request merged**